### ISSUE TITLE
Error: package or namespace load failed for ‘tidyverse’

### ISSUE THREAD

**Issue by donojazz** [tidyverse/tidyverse#316]
I have been wrestling with a marmap function [getNOAA.bathy()] over the last two days.  Their sample code does not run ,  They claimed it was my problem and said to reload my R (4.2.2) and R Studio (2022.07.2 Build 576).  So I did.   Now suddently my Tidyverse library will not load. I have deleted and reloaded several libraries (rlang, rccp, and tidyverse) to no avail. Can you help?

This is error wessage:
______
Error: package or namespace load failed for ‘tidyverse’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
namespace ‘vctrs’ 0.4.1 is already loaded, but >= 0.5.0 is required
______

Traceback:

Error: package or namespace load failed for ‘tidyverse’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
namespace ‘vctrs’ 0.4.1 is already loaded, but >= 0.5.0 is required
6.
stop(msg, call. = FALSE, domain = NA)
5.
value[[3L]](cond)
4.
tryCatchOne(expr, names, parentenv, handlers[[1L]])
3.
tryCatchList(expr, classes, parentenv, handlers)
2.
tryCatch({
attr(package, "LibPath") <- which.lib.loc
ns <- loadNamespace(package, lib.loc)
env <- attachNamespace(ns, pos = pos, deps, exclude, include.only) ...
1.
library(tidyverse)
______

_____
Here is my system info as it sitr right now:

R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.4.1    tigris_1.6.1    sf_1.0-9        tmaptools_3.1-1 tmap_3.3-3     

loaded via a namespace (and not attached):
 [1] httr_1.4.4         tidyr_1.2.1        viridisLite_0.4.1  assertthat_0.2.1   sp_1.5-1           cellranger_1.1.0  
 [7] yaml_2.3.6         pillar_1.8.1       backports_1.4.1    lattice_0.20-45    glue_1.6.2         uuid_1.1-0        
[13] digest_0.6.29      RColorBrewer_1.1-3 colorspace_2.0-3   htmltools_0.5.3    XML_3.99-0.12      pkgconfig_2.0.3   
[19] broom_1.0.1        raster_3.6-3       stars_0.5-6        purrr_0.3.5        scales_1.2.1       terra_1.6-17      
[25] tibble_3.1.8       proxy_0.4-27       generics_0.1.3     leafsync_0.1.0     cli_3.4.0          magrittr_2.0.3    
[31] maptools_1.1-5     evaluate_0.18      fansi_1.0.3        forcats_0.5.2      lwgeom_0.2-9       foreign_0.8-83    
[37] class_7.3-20       tools_4.2.2        lifecycle_1.0.3    stringr_1.4.1      munsell_0.5.0      compiler_4.2.2    
[43] e1071_1.7-12       rlang_1.0.6        classInt_0.4-8     units_0.8-0        grid_4.2.2         dichromat_2.0-0.1 
[49] rstudioapi_0.14    rappdirs_0.3.3     htmlwidgets_1.5.4  crosstalk_1.2.0    leafem_0.2.0       base64enc_0.1-3   
[55] rmarkdown_2.18     gtable_0.3.1       codetools_0.2-18   abind_1.4-5        DBI_1.1.3          R6_2.5.1          
[61] knitr_1.40         dplyr_1.0.10       rgdal_1.6-2        fastmap_1.1.0      utf8_1.2.2         KernSmooth_2.23-20
[67] stringi_1.7.8      parallel_4.2.2     Rcpp_1.0.9         vctrs_0.4.1        png_0.1-7          leaflet_2.1.1     
[73] dbplyr_2.2.1       tidyselect_1.2.0   xfun_0.33  
_____

Not sure where to go right I love the Tidyverse and will be lost without it.   Thanks  

Jim

**Comment by donojazz** [comment]
Re-Re loaded every thing but problem is back.

**Comment by donojazz** [comment]
Added error message piece after re-installs

Registered S3 methods overwritten by 'dbplyr':
  method         from
  print.tbl_lazy     
  print.tbl_sql      
Error: package or namespace load failed for ‘tidyverse’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 namespace ‘vctrs’ 0.4.1 is already loaded, but >= 0.5.0 is required

**Comment by jennybc** [comment]
This will probably help you understand what you're going through and how to fix and how to prevent: https://github.com/rstats-wtf/what-they-forgot/issues/62

**Comment by donojazz** [comment]
Thanks, Jenny.

I thought I had pretty much followed those procedures.  Closed R, R Studio, removed all instances of programs with the windows app uninstall and then rebooted. Then, I reloaded fresh versions (absolute latest) of R and RStudio.  I did get tidyverse to run cleanly one time and then it reverted to the error.  The only thing else I can think of now is to erase ALL of the libraries in the 4.2.2 folder.  I did not do that yet.

Should I absolutely flush out everything, reboot and start afresh?  A recommentation would be much appreciated, especially from you.  I am participating in the 30Day Challenge so I ave been creating/snarfing/copying geospatial code bits from lots of sources.  That may have been the culprit.  I'm only at day 18 and I really need tidyverse -  love it.  Jim

**Comment by jennybc** [comment]
>  I did get tidyverse to run cleanly one time and then it reverted to the error

It's hard to say what's going on but I guess I'd do a thorough check of all your libraries (meaning: the place where packages get installed) and make sure nothing you see surprises you. And if you find multiple libraries, which you will by default on windows, see if vctrs is installed in more than one at different versions?

**Comment by donojazz** [comment]
I found 7 vctrs folders filled with files.  They were in each R version folder spannin 4.0 to 4.2.  Five were from version 4.1.

I think I am going to totally purge everything associated with R and start afresh with the latest R and RStudio versions.

I will keep you posted on the outcome.  Thanks.    Jim

**Comment by donojazz** [comment]
Jenny,  I completely purged anything R anywhere;  THere were sone files that went back to 3,5.  The good new is that Tidryverse is running.  Now I have to rebuild my library stash.

I am have some issues with Tmap, but will have to play with it some more.  Something about file polygons not right.  However thes are census shp files and ran well before.  I may be missing a library.

I have a question on the latest RStudio.  Should I let it install the missing libraries after I load my code?  Is that the same as  install.packages() ?  That was frowned on by lionel -  https://github.com/rstats-wtf/what-they-forgot/issues/62#issuecomment-545038970     Is that a true statement.  I have done that in the past with many of my libraries.

jim

**Comment by jennybc** [comment]
I would probably take a pragmatic approach of taking the path of least resistance most of the time (using `install.packages()`, accepting RStudio's invitation to install packages, etc.). And keep https://github.com/rstats-wtf/what-they-forgot/issues/62#issuecomment-545038970 bookmarked for getting things unborked when you have the problem of a broken package install.

But Windows is not my primary OS. Maybe some people really do quit every running R session before they install packages? And do so in a `--vanilla` session? I can't say.

Finally, personally, I use pak for most of my package installation, which is VASTLY better about this problem on Windows:

https://pak.r-lib.org

I guess it's still considered somewhat beta, in the sense that we haven't officially recommended this over `install.packages()`. But pak is especially compelling, today, for Windows users.

**Comment by donojazz** [comment]
OK.  I finally fixed tidyverse by cleaning up old R files and 7 vctrs files.  Then a re-reinstallation of all software.

My tmap issue were solved when I discovered RStudio was not loading geos when I called it up.  So I used the console with instali.packages("geos").  This installed the missing libraries:

package ‘libgeos’ successfully unpacked and MD5 sums checked
package ‘wk’ successfully unpacked and MD5 sums checked
package ‘geos’ successfully unpacked and MD5 sums checked

and tmap ran.   https://github.com/r-spatial/sf/issues/2043#issuecomment-1320689150

So, all is good in Mudville and my software is now working well.  I can get back to my 30 Day Map Challenge using both tidyverse and tmap.

Thanks for all the help.  Another great learning situation!!   Jim
